### ISSUE TITLE
Settings - "Microsoft ID" and "Remember login credentials" don't work

### ISSUE THREAD

**Issue by muzso** [kylon/Sharedown#58]
I'm using the latest state of master branch (commit: 45c908483d0c78185e0564d6d415a87ae48b9f8a).

If I enter something in Settings in the "Microsoft ID" field, press "Save" and restart the app, the field is empty again.

I've checked the "Remember login credentials" checkbox and tried to download multiple videos (without restarting the app!) and after each video I'm asked for credentials again and again. Of course the same applies if I restart the app.

I've checked the OS provided credential store (in my case Ubuntu uses the Seahorse app for the UI of password/credentials management) and there's no sign of any credentials saved by Sharedown.

I can provide logs on request, but my guess is that this is trivial to reproduce (at least on Ubuntu ... the issue might be platform-specific).

**Comment by kylon** [comment]
> If I enter something in Settings in the "Microsoft ID" field, press "Save" and restart the app, the field is empty again.

this is not saved since it is considered sensitive data, it is not written in settings file (i think this is mentioned in settings too)



> I've checked the "Remember login credentials" checkbox and tried to download multiple videos (without restarting the app!) and after each video I'm asked for credentials again and again. Of course the same applies if I restart the app.

sounds like you need `keep chrome window open` workaround if it asks for 2fa codes or similar
it should work if **id**, **name** and **password** are enough to complete the login



> I've checked the OS provided credential store (in my case Ubuntu uses the Seahorse app for the UI of password/credentials management) and there's no sign of any credentials saved by Sharedown.

i use arch, and i have seahorse

![2022-12-03-214140_594x122_scrot](https://user-images.githubusercontent.com/3252255/205461360-8d76361b-53b2-43e4-99bf-628b18498921.png)

![2022-12-03-214238_560x163_scrot](https://user-images.githubusercontent.com/3252255/205461362-959b29eb-5140-4eed-a8f9-0ce5782de8e7.png)

sharedown saves its data in default keyring

now, there seem to be a bug where ms id is not filled if basic login module is selected, i ll fix this

check your default keyring and that everything is set correctly in ubuntu

if the problem persists, i ll setup a vm (what build are you using?)

**Comment by muzso** [comment]
> this is not saved since it is considered sensitive data, it is not written in settings file
> (...)
> unless you save the credentials

But I did check "Remember login credentials" checkbox.
Or do you mean something else by "save the credentials"?

> i think this is mentioned in settings too

I don't see this mentioned.

> sounds like you need `keep chrome window open` workaround if it asks for 2fa codes or similar it should work if **id**, **name** and **password** are enough to complete the login

I do have 2fa set up.
The "Keep Chrome window open" option is greyed out for me, I cannot seem to be able to enable it.

> i use arch, and i have seahorse
> (...)
> sharedown saves its data in default keyring

In Ubuntu the default keyring (i.e. the one that is automatically unlocked during user login) is called "Login".
And I did have Sharedown entries there, but I don't know when they were created. Probably sometime in the past with an older Sharedown version. I've removed them and ever since regardless of what I do (in Settings), the Sharedown credentials don't appear anymore. I can test this with an older version to see where it broke for me.

> now, there seem to be a bug where ms id is not filled if basic login module is selected, i ll fix this

I use the default login module ("Basic - Manual login"), so the bug might be the cause why saving the credentials to the keyring doesn't work for me.

> check your default keyring and that everything is set correctly in ubuntu

There're a lot of entries from other apps, so everything seems to be set up correctly as far as the OS-side is concerned.

> if the problem persists, i ll setup a vm (what build are you using?)

Build? You mean Ubuntu?

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal
```

It's of course updated with all available patches.

**Comment by kylon** [comment]
> I do have 2fa set up.
> The "Keep Chrome window open" option is greyed out for me, I cannot seem to be able to enable it.

it conflicts with remember credentials, so disable that to enable it

> But I did check "Remember login credentials" checkbox.
> Or do you mean something else by "save the credentials"?

when remember is not selected

i created that entry to test it, so with latest version


> Build? You mean Ubuntu?

yes, i ll fix the ms id bug first and, if it does not solve the issue, i ll install that and see what happen

**Comment by kylon** [comment]
so, master fixes the ms id bug (actually a mistake)
about ubuntu, i tested it and everything is working

i also tried to manually delete the entry and it was able to recreate it

![Screenshot from 2022-12-11 13-07-52](https://user-images.githubusercontent.com/3252255/206902831-c6199b9e-3736-4b22-b792-c0e8b04cdddb.png)
![Screenshot from 2022-12-11 13-08-08](https://user-images.githubusercontent.com/3252255/206902833-6cfd546d-a442-443a-afae-bde40aa76ed0.png)

**Comment by kylon** [comment]
closing for inactivity

**Comment by muzso** [comment]
Sorry for not responding so long.
I've tested with the just released v5.3.2 and "misd" is saved now into the default keyring ("Login" keyring). Thanks!
It works too, i.e. upon Sharedown restart Chrome doesn't ask for the username.

Password is still not saved with "Remember login credentials" enabled and using the "Basic - manual login" login module.
But in your screenshot it's the same for you too. I only see an "msid" on your screenshot, no password.

**Comment by kylon** [comment]
basic login has no credentials, so nothing to save, is this the one you are trying to use with keyring?

**Comment by muzso** [comment]
> basic login has no credentials, so nothing to save, is this the one you are trying to use with keyring?

That explains a lot! :-)
I always thought that the "Remember login credentials" meant the password that I entered into Chrome while logging into the account. But now I realized from your answer that if I change the login method to something else (in Sharedown's Settings), a password input field appears (in Settings) and that's the credential that Sharedown can record in the keyring. Thanks for clearing up my confusion.
